chiselFrontend/src/main/scala/chisel3/core/MultiClock.scala

```diff
@@ -48,7 +56,15 @@ object withReset {  // scalastyle:ignore object.name
     * @param block the block of code to run with new implicit Reset
     * @return the result of the block
     */
-  def apply[T](reset: Reset)(block: => T): T =
-    withClockAndReset(Module.clock, reset)(block)
+  def apply[T](reset: Reset)(block: => T): T = {
+    // Save parentScope
+    val parentReset = Builder.currentReset
+    Builder.currentReset = Builder.currentReset
```

Comment:
I believe this should be
```suggestion
    Builder.currentReset = Some(reset)
```

Reply:
Good catch @jackkoenig, thanks!